hack/verify-staticcheck.sh

```diff
@@ -5,15 +5,26 @@ set -o nounset
 set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-GOLANGCI_LINT_PKG="github.com/golangci/golangci-lint/cmd/golangci-lint"
-GOLANGCI_LINT_VER="v1.32.2"
+GOLANGCI_LINT_VER="v1.40.1"
 
 cd "${REPO_ROOT}"
 source "hack/util.sh"
 
-util::install_tools ${GOLANGCI_LINT_PKG} ${GOLANGCI_LINT_VER}
+# directly download binary, details refer to https://golangci-lint.run/usage/install
+curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin ${GOLANGCI_LINT_VER}
```

Comment:
Why? For performance?

Reply:
Mostly, details see  https://golangci-lint.run/usage/install

Reply:
Can we skip the download if the binary already exists?

Reply:
sure